### PR TITLE
Update arguments and attributes in README.md

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

**Comment by soerenmartius** [comment]
[CH735]

File: README.md [hunk 1]
```diff
@@ -46,6 +44,9 @@ A [Terraform] 0.12 module for managing an organization on [Github].
 
 ## Getting Started
 
+The GitHub Provider requires a GitHub Token and the name of the organization.
+You can provide your credentials via the `GITHUB_TOKEN` and `GITHUB_ORGANIZATION` environment variables.
+
```
Comment (mariux) [inline README.md:49]: this is not the scope of the module to care about provider configuration.. as people can use all the ways to configure this when they define the provider.. so if the provider picks up the same config of gh-cli this is even not needed at all.

File: README.md [hunk 2]
```diff
@@ -85,9 +86,33 @@ See [variables.tf] and [examples/] for details and use-cases.
 
 ### Top-level Arguments
 
-#### Main Resource Configuration
+- **`blocked_users`**: *(Optional `set(string)`)*
+  A list of usernames to be blocked from a GitHub organization.
```
Comment (mariux) [inline README.md:90]: empty lines are missing

File: README.md [hunk 3]
```diff
@@ -85,9 +86,33 @@ See [variables.tf] and [examples/] for details and use-cases.
...
+  Default is `[]`.
+
+- **`admins`**: *(Optional `set(string)`)*
+  A list of users to be added to your organization with admin role.
+  When applied, an invitation will be sent to the user to become part of the organization.
+  When destroyed, either the invitation will be cancelled or the user will be removed.
+  Default is `[]`.
+
+- **`projects`**: *(Optional `list(any)`)*
+  Create and manage projects for the GitHub organization.
```
Comment (mariux) [inline README.md:106]:
```suggestion
- **`projects`**: *(Optional `list(project)`)*

  Create and manage projects for the GitHub organization.
```

File: variables.tf [hunk 4]
```diff
@@ -17,54 +17,60 @@
 # ---------------------------------------------------------------------------------------------------------------------
 
 variable "all_members_team_name" {
-  description = "The name of the team that contains all members of the organization."
+  description = "(Optional) The name of the team that contains all members of the organization."
   type        = string
   default     = null
```
Comment (mariux) [inline variables.tf:22]: what is the actual default?

File: variables.tf [hunk 5]
```diff
@@ -17,54 +17,60 @@
...
 }
 
 variable "all_members_team_visibility" {
-  description = "The level of privacy for the team. Must be one of `secret` or `closed`."
+  description = "(Optional) The level of privacy for the team. Must be one of `secret` or `closed`."
```
Comment (mariux) [inline variables.tf:26]:
```suggestion
  description = "(Optional) The level of privacy for the team. Must be one of `secret` or `closed`. Default is \"secret\""
```

File: variables.tf [hunk 6]
```diff
@@ -17,54 +17,60 @@
...
   type        = string
   default     = "secret"
 }
 
 variable "blocked_users" {
+  description = "(Optional) A list of usernames to be blocked from a GitHub organization."
```
Comment (mariux) [inline variables.tf:32]:
```suggestion
  description = "(Optional) A list of usernames to be blocked from a GitHub organization. Default is []."
```

File: variables.tf [hunk 7]
```diff
@@ -17,54 +17,60 @@
...
   # ]
+  #
 
   default = []
 }
 
 variable "members" {
   type        = set(string)
-  description = "A list of users to be added to your organization with member role. When applied, an invitation will be sent to the user to become part of the organization. When destroyed, either the invitation will be cancelled or the user will be removed."
+  description = "(Optional) A list of users to be added to your organization with member role. When applied, an invitation will be sent to the user to become part of the organization. When destroyed, either the invitation will be cancelled or the user will be removed."
```
Comment (mariux) [inline variables.tf:48]:
```suggestion
  description = "(Optional) A list of users to be added to your organization with member role. When applied, an invitation will be sent to the user to become part of the organization. When destroyed, either the invitation will be cancelled or the user will be removed. Default is []."
```

File: variables.tf [hunk 8]
```diff
@@ -17,54 +17,60 @@
...
   # ]
+  #
 
   default = []
 }
 
 variable "admins" {
   type        = set(string)
-  description = "A list of users to be added to your organization with admin role. When applied, an invitation will be sent to the user to become part of the organization. When destroyed, either the invitation will be cancelled or the user will be removed."
+  description = "(Optional) A list of users to be added to your organization with admin role. When applied, an invitation will be sent to the user to become part of the organization. When destroyed, either the invitation will be cancelled or the user will be removed."
```
Comment (mariux) [inline variables.tf:64]:
```suggestion
  description = "(Optional) A list of users to be added to your organization with admin role. When applied, an invitation will be sent to the user to become part of the organization. When destroyed, either the invitation will be cancelled or the user will be removed. Default is []."
```

File: variables.tf [hunk 9]
```diff
@@ -79,8 +85,10 @@ variable "projects" {
   #   name = string
   #   body = optional(string)
   # }))
-  description = "Create and manage projects for the GitHub organization."
 
+  description = "(Optional) Create and manage projects for the GitHub organization."
```
Comment (mariux) [inline variables.tf:89]:
```suggestion
  description = "(Optional) Create and manage projects for the GitHub organization. Default is []."
```

File: variables.tf (hunk 4 above)
Comment (soerenmartius) [reply variables.tf:22]: null

File: variables.tf (hunk 5 above)
Comment (soerenmartius) [reply variables.tf:26]: We don't need that as the default value is already set. No reason to repeat what your IDE is showing you already anyways..

File: variables.tf (hunk 6 above)
Comment (soerenmartius) [reply variables.tf:32]: same here

File: variables.tf (hunk 7 above)
Comment (soerenmartius) [reply variables.tf:48]: and same here

File: variables.tf (hunk 8 above)
Comment (soerenmartius) [reply variables.tf:64]: same here

File: README.md (hunk 3 above)
Comment (soerenmartius) [reply README.md]: resolved in c2d1bab

File: variables.tf (hunk 9 above)
Comment (soerenmartius) [reply variables.tf:89]: and same here

File: README.md (hunk 1 above)
Comment (soerenmartius) [reply README.md]: resolved in c2d1bab

File: README.md (hunk 2 above)
Comment (soerenmartius) [reply README.md]: resolved in c2d1bab

File: README.md [hunk 10]
```diff
@@ -97,13 +124,24 @@ The following attributes are exported by the module:
...
+- **`projects`**
 
-  The full `resource` object with all its attributes.
+  A list of `github_organization_project` resource objects that
+  describe all projects of the organization.
 
 ## External Documentation
 
-- Terraform Github Provider Documentation:
+- Terraform Github Provider Documentation: https://www.terraform.io/docs/providers/github/index.html
```
Comment (thenaturalist) [inline README.md:144]:
```suggestion
- Terraform Github Provider Documentation: 
   - https://www.terraform.io/docs/providers/github/index.html
```

File: variables.tf (hunk 5 above)
Comment (mariux) [reply variables.tf]: Terradoc will create variables.tf not read it..

Comment (soerenmartius) [reply variables.tf]: Yes correct, still we don't need it in the description as well.

File: variables.tf [hunk 11]
```diff
@@ -17,54 +17,60 @@
...
   type        = string
   default     = null
 }
 
 variable "all_members_team_visibility" {
-  description = "The level of privacy for the team. Must be one of `secret` or `closed`."
+  description = "(Optional) The level of privacy for the team. Must be one of 'secret' or 'closed'."
   type        = string
   default     = "secret"
 }
```
Comment (mariux) [inline variables.tf:29]: which default is more secure?

Comment (soerenmartius) [reply variables.tf:29]: `Secret` - From the docs:

> Secret teams are only visible to the people on the team and people with owner permissions. They're great for hiding teams with sensitive names or members, such as those used for working with external partners or clients. Secret teams cannot be nested under parent teams or have child teams.

**Pull request merged**